dvc/api.py

```diff
@@ -1,15 +1,28 @@
+from contextlib import contextmanager
+
+from dvc.utils.compat import urlparse
 from dvc.repo import Repo
+from dvc.pkg.pkg import temp_repo
 
 
 def get_url(path, repo_dir=None, remote=None):
     """Returns an url of `path` in default or specified remote"""
-    repo = Repo(repo_dir)
-    out, = repo.find_outs_by_path(path)
-    remote_obj = repo.cloud.get_remote(remote)
-    return str(remote_obj.checksum_to_path_info(out.checksum))
+    with _make_repo(repo_dir) as repo:
+        out, = repo.find_outs_by_path(path)
+        remote_obj = repo.cloud.get_remote(remote)
+        return str(remote_obj.checksum_to_path_info(out.checksum))
 
 
 def open(path, repo_dir=None, remote=None, mode="r", encoding=None):
     """Opens a specified resource as a file descriptor"""
-    repo = Repo(repo_dir)
-    return repo.open(path, remote=remote, mode=mode, encoding=encoding)
+    with _make_repo(repo_dir) as repo:
+        return repo.open(path, remote=remote, mode=mode, encoding=encoding)
+
+
+@contextmanager
+def _make_repo(repo_dir):
+    if urlparse(repo_dir).scheme == "":
```

Comment:
what about `file:` scheme? just out of curiosity, not sure if we should care about it now

Reply:
No point in using `file://` scheme you can just use path. We don't use scheme for local paths anywhere in dvc.